docker.sh

```diff
@@ -40,6 +40,7 @@ build(){
         --build-arg RUST_TOOLCHAIN=${RUST_TOOLCHAIN} \
         --build-arg GIT_BRANCH="${GIT_BRANCH}" \
         --build-arg GIT_COMMIT="${GIT_COMMIT}" \
+        --build-arg ARCH="${ARCH}" \
```

Comment:
How is this argument created? does it need to be passed when running `./docker.sh`?

Reply:
@andreeaflorescu this is already set at the top of this file:

ARCH=$(uname -m)

I think the first cleanup patch still makes sense, we can drop the musl thing for now.